src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedCompactor.scala

```diff
@@ -33,12 +33,14 @@ object SegmentedCompactor {
     val maxDeleteAgeHours = siriusConfig.getInt(SiriusConfiguration.COMPACTION_MAX_DELETE_AGE_HOURS, Integer.MAX_VALUE)
     val maxDeleteAgeMillis = 60L * 60 * 1000 * maxDeleteAgeHours
 
-    new SegmentedCompactor(maxDeleteAgeMillis)
+    new SegmentedCompactor(maxDeleteAgeMillis, siriusConfig)
   }
 }
 
-private [segmented] class SegmentedCompactor(maxDeleteAgeMillis: Long) {
+private [segmented] class SegmentedCompactor(maxDeleteAgeMillis: Long, siriusConfig: SiriusConfiguration) {
 
+  def segment(base: JFile, name: String): Segment = Segment(siriusConfig, base, name)
+  def segment(location: JFile): Segment = Segment(siriusConfig, location)
```

Comment:
I don't love having to weave the config into the class instantiation here and pass it through to our `Segment` instantiation.

The alternative would be to have apply pass a `buildSegment: (JFile => Segment)` as a parameter to the `SegmentedCompactor` class. I might actually prefer that tbh, as it keeps all the configuration mess in the `apply` function.

Like:
```
  def apply(siriusConfig: SiriusConfiguration) : SegmentedCompactor = {
    def buildSegment(file: JFile): Segment = Segment(siriusConfig, file)

    val maxDeleteAgeHours = siriusConfig.getInt(SiriusConfiguration.COMPACTION_MAX_DELETE_AGE_HOURS, Integer.MAX_VALUE)
    val maxDeleteAgeMillis = 60L * 60 * 1000 * maxDeleteAgeHours

    new SegmentedCompactor(maxDeleteAgeMillis, buildSegment)
  }
```

```
private [segmented] class SegmentedCompactor(maxDeleteAgeMillis: Long, buildSegment: (JFile => Segment)) {
  def buildSegment(base: JFile, name: String): Segment = buildSegment(new JFile(base, name))
  // ...
```

Also I'd generally prefer `buildSegment` to `segment` as far as naming is concerned, just to make it a little clearer what's happening.